apps/wallet/src/ui/app/pages/approval-request/utils.ts

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// ported from https://github.com/hcodes/isutf8
+
+export function isUtf8(buf?: Buffer | Uint8Array): boolean {
```

Comment:
is there no better way to detect this 😆, e.g. try to decode the string as UTF8 with `TextDecoder` and if it doesn't work then fall back to returning the string in base64?

Reply:
Thanks updated it